### PR TITLE
Jwt security

### PR DESCRIPTION
**Pull request by mkazarian-1**

### CODE REVIEW

File: src/main/java/org/example/bookshop/security/JwtAuthorizationFilter.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+                    userDetails, null, userDetails.getAuthorities()
+            );
+            SecurityContextHolder.getContext().setAuthentication(authorization);
+        }
+
+        filterChain.doFilter(request, response);
+    }
+
+    private String getToken(HttpServletRequest request) {
+        String bearerToken = request.getHeader("Authorization");
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookshop/security/JwtAuthorizationFilter.java]:
```suggestion
        String bearerToken = request.getHeader(HttpHeaders.AUTHORIZATION);
```

File: src/main/java/org/example/bookshop/security/JwtAuthorizationFilter.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        if (StringUtils.hasText(bearerToken) && bearerToken.startsWith("Bearer ")) {
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookshop/security/JwtAuthorizationFilter.java]: create constant for "Bearer "

File: src/main/java/org/example/bookshop/security/JwtAuthorizationFilter.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+            return bearerToken.substring(7);
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookshop/security/JwtAuthorizationFilter.java]:
```suggestion
            return bearerToken.substring(constant.length());
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

File: src/main/java/org/example/bookshop/config/SecurityConfig.java [hunk 4]
```diff
@@ -30,15 +35,25 @@ public SecurityFilterChain securityFilterChain(HttpSecurity httpSecurity) throws
                 .cors(AbstractHttpConfigurer::disable)
                 .csrf(AbstractHttpConfigurer::disable)
                 .authorizeHttpRequests(auth -> auth
-                        .requestMatchers("/auth/**").permitAll()
-                        .requestMatchers("/swagger-ui/**", "/v3/api-docs/**").permitAll()
+                        .requestMatchers("/auth/**")
+                        .permitAll()
+                        .requestMatchers("/swagger-ui/**", "/v3/api-docs/**")
+                        .permitAll()
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookshop/config/SecurityConfig.java]: combine into one requestMatchers

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
one comment

File: src/main/java/org/example/bookshop/dto/user/UserLoginRequestDto.java [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.Pattern;
+import lombok.Getter;
+import lombok.Setter;
+
+@Getter
+@Setter
+public class UserLoginRequestDto {
+    @NotBlank
+    @Pattern(regexp = "^[^@]+@[^@]+\\.[^@]+$")
```
Comment (ahoienko) [inline src/main/java/org/example/bookshop/dto/user/UserLoginRequestDto.java]:
```suggestion
    @NotBlank
    @Email
```

**Review by ahoienko — APPROVED** [review]
Good job! Please check minor comment

**Pull request merged**